BossMod/BossModule/BossComponent.cs

```diff
@@ -39,6 +39,10 @@ public virtual void Update() { } // called every frame - it is a good place to u
     public virtual void AddMovementHints(int slot, Actor actor, MovementHints movementHints) { } // gather movement hints for specified raid member
     public virtual void AddGlobalHints(GlobalHints hints) { } // gather any relevant pieces of advice for whole raid
     public virtual void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints) { } // gather AI hints for specified raid member
+    public virtual void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints, float maxCastTime)
```

Comment:
Why is it needed? I don't like the idea of adding cruft to such a low level interface, if this is unavoidable, i'd rather just change the API and do a huge search & replace job - but first I'd like to understand the intent...

Reply:
this is to make cast time calculation work in fully automatic modules, i.e. RP duties. IMO it makes a lot more sense to have a module handle the rotation in its entirety since a specific RP character can't be played outside that duty, meaning there's no point in writing a separate rotation for it.

technically you can just pass float.MaxValue in the replay viewer which is what I did, i think we already do that for rotations as well

Reply:
Should no longer be needed, this field is now in AIHints.

Reply:
removed